test/unit-test/CMakeLists.txt

```diff
@@ -6,9 +6,26 @@ set(project_name "core_json")
 # =====================  Create your mock here  (edit)  ========================
 # ================= Create the library under test here (edit) ==================
 
+# Base name for temporary files
+set( TEMP_BASE ${CMAKE_BINARY_DIR}/${project_name} )
+
+# Strip static constraints so unit tests may call internal functions
+execute_process( COMMAND sed "s/^static //"
+                 WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+                 INPUT_FILE ${JSON_SOURCES}
+                 OUTPUT_FILE ${TEMP_BASE}.c
+        )
+
+# Generate a header file for internal functions
+execute_process( COMMAND sed -n "1s/.*/typedef int bool_;/p; /^static.*(/,/^{\$/{s/^static //; s/)\$/&;/; /{/d; p;}"
+                 WORKING_DIRECTORY ${CMAKE_BINARY_DIR}

```

Comment:
Maybe have this be `${CMAKE_BINARY_DIR}/headers` or something within those lines

Reply:
I feel like these two generated files should be front and center since they're what's actually being compiled, and they are automatically generated so I want to spot check from time to time.